lib/govuk_design_system_formbuilder/elements/date.rb

```diff
@@ -72,7 +72,15 @@ def date_part(segment, width:, link_errors: false)
       def value(segment)
         attribute = @builder.object.try(@attribute_name)
 
-        attribute.try(segment) || attribute.try(:[], MULTIPARAMETER_KEY[segment])
+        return unless attribute
+
+        if attribute.respond_to?(segment)
+          attribute.send(segment)
+        elsif attribute.respond_to?(:fetch)
+          attribute.fetch(MULTIPARAMETER_KEY[segment])
```

Comment:
I'm wondering if using `fetch` is too strict, if someone defines a multi-parameter attribute that does not include the correct segment do we want to ignore it and return `nil` instead or throw a `key not found (KeyError)`?

Reply:
Personally I'd _want_ failure here, but I can see how everyone might not like that.

I amended the behaviour a bit so it'll fire a warning message if nothing's fetched but will return `nil`.

I'm going to add a subsequent PR that wraps `Rails.logger.warn` so they can be toggled on/off library-wide.